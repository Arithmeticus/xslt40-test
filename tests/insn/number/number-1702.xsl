<?xml version="1.0"?> 
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="2.0">

<?spec xslt#current-function?>
<!-- Test use of current() in the from pattern of xsl:number -->

<xsl:output indent="yes"/>

<xsl:template match="*">
  <xsl:copy>
    <xsl:attribute name="nr">
      <!-- count distance from the last node having the same name as its parent -->
      <xsl:number from="*[name()=name(current())]/*" level="any"/>
    </xsl:attribute>
    <xsl:apply-templates/>
  </xsl:copy>
</xsl:template>
 
</xsl:stylesheet>
