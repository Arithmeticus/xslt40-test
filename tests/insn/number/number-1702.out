<?xml version="1.0" encoding="UTF-8"?>
<doc nr="1">
  <a nr="1">
      <a nr="1"/>
  </a>
  <a nr="2">
      <b nr="1"/>
  </a>
  <a nr="3">
      <a nr="1"/>
  </a>
  <a nr="2">
      <b nr="1"/>
      <b nr="2"/>
  </a>  
</doc>